src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -440,6 +461,33 @@ private static void AddProcessRegistry(IServiceCollection services)
             }
         }
 
+        private static IDistributedLockManager GetBlobLockManagerOverride(IServiceProvider provider, ScriptApplicationHostOptions applicationHostOptions)
+        {
+            try
+            {
+                //if (applicationHostOptions.HasParentScope)
+                //{
+                //    provider = applicationHostOptions.RootServiceProvider;
+                //}
+
+                var azureStorageProvider = provider.GetRequiredService<IAzureStorageProvider>();
+                var container = azureStorageProvider.GetBlobContainerClient();
+
+                if (container != null)
+                {
+                    var loggerFactory = provider.GetRequiredService<ILoggerFactory>();
+                    return new BlobLeaseDistributedLockManager(loggerFactory, azureStorageProvider);
+                }
+            }
```

Comment:
Are you expecting exceptions in this case? We shouldn't be swallowing it here. Are the operations being introduced above expensive?

Reply:
This logic could throw an InvalidOperationException (if no valid connection), UriFormatException (if the sas container override is not properly formatted), or ArgumentException (validation logic in creating a BlobContainerClient at the sdk level). Thought the default behavior should be to register an InMemoryDistributedLockManager in this case.

Reply:
All of the above indicate an improperly configured app and this behavior will hide the issue from the user, applying a fallback that would likely confuse them and potentially cause the configuration issue to go unnoticed. Can you please update this so we're not hiding the problem and swallowing the exception?